### PR TITLE
Remove remoting2 dependency

### PR DESCRIPTION
**Pull request by pkoenig10**
**Goals (and why)**:

Eliminate outdated dependencies.

I think most of the hard work was done in https://github.com/palantir/atlasdb/pull/4561.

**Implementation Description (bullets)**:

- Remove `ExceptionMappers`

    Exception responses no longer have bodies. Not exposing exception class names and stacktraces in response bodies is a positive outcome of this change.

**Testing (What was existing testing like?  What have you done to improve it?)**:

- Strengthen some assertions in tests with comments referencing `remoting2`, effectively reverting the changes in https://github.com/palantir/atlasdb/commit/bee5aa4825f4fd2d41cb62f19be2cc7cb2415396

    This PR does not change the behavior in those tests. The test changes could be made without removing `remoting2`.
 
**Concerns (what feedback would you like?)**:

- Is it safe to remove the exception response bodies?

     These are QoS exceptions (429, 503, etc.) so my expectation is that the response body is purely informational and thus safe to remove. This matches how [Conjure handles QoS exceptions](https://github.com/palantir/conjure-java/blob/e3680f29fb52e79d0223848468d1e01166554dac/conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptions.java#L97).

### CODE REVIEW

**Comment by jeremyk-91** [comment]
Hey @pkoenig10, thanks for the PR. I think we were initially cautious about this because this change affects timelock _server_, and the risk is that there may be some random client / user of timestamp and lock services that was written _before_ AtlasDB moved to using Conjure. I'll write a small PR to get metrics; if that reveals no one is actually using V1 any more, we can go ahead with this.

**Comment by pkoenig10** [comment]
And the concern is that those clients are using something other than CJR or Dialogue clients? Because both of those clients handle QoS response status codes without using the body.
- https://github.com/palantir/conjure-java-runtime/blob/d02dcf5e58da34f95bca521120f78d4687d979c8/okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/RemotingOkHttpCall.java#L340-L345
- https://github.com/palantir/conjure-java/blob/e3680f29fb52e79d0223848468d1e01166554dac/conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptions.java#L97

**Comment by jeremyk-91** [comment]
Correct. The concern is about clients that pre-date the time we switched to use CJR.

**Comment by pkoenig10** [comment]
@jeremyk-91 Taking a look at logs over the last week, all logged user-agents are dialogue clients. So I'm fairly confident that this change is safe.

**Comment by jeremyk-91** [comment]
Yep, thanks - confirmed that's the case. Strange that we'd serve them the legacy thing nonetheless (it means people are making their *own* `JaxRsClient` and talking to timelock in that way - which I'm aware some of the products do, though wasn't aware of all of the products in that list), but it looks like they're ignoring the response body anyway...

**Comment by jeremyk-91** [comment]
👍

File: timelock-impl/src/main/java/com/palantir/atlasdb/timelock/TooManyRequestsExceptionMapper.java [hunk 1]
```diff
@@ -15,17 +15,15 @@
...
 import javax.ws.rs.ext.ExceptionMapper;
 
 public class TooManyRequestsExceptionMapper implements ExceptionMapper<TooManyRequestsException> {
-    public static final int TOO_MANY_REQUESTS_429 = 429;
 
     @Override
     public Response toResponse(TooManyRequestsException exception) {
-        return ExceptionMappers.encodeExceptionResponse(exception, TOO_MANY_REQUESTS_429)
-                .build();
+        return Response.status(Status.TOO_MANY_REQUESTS).build();
```
Comment (gsheasby) [inline timelock-impl/src/main/java/com/palantir/atlasdb/timelock/TooManyRequestsExceptionMapper.java:27]: 👍

**Review by gsheasby — APPROVED** [review]

**Pull request merged**